### ISSUE TITLE
Rounding Errors - TruncateInt(), RoundInt()  vs Ceil().TruncateInt() usage: Swap

### ISSUE THREAD

**Issue by jcompagni10** [duality-labs/duality#313]
TruncateInt(), Sub usages - The starting point in our analysis was the calculation in:
Swap functions for pool liquidity swaps -  [swap0to1](https://github.com/duality-labs/duality/blob/377592adb4ac0ef445c5e2bc2a73e2635189ccfc/x/dex/keeper/pool.go#L66-L69) and [swap1to0](https://github.com/duality-labs/duality/blob/377592adb4ac0ef445c5e2bc2a73e2635189ccfc/x/dex/keeper/pool.go#L94-L97)
Where we analyzed the impact of doing TruncateInt() and then subtraction of that amount in order to calculate the remaining reserves in a tick. Since TruncateInt() rounds down, the amount of reserves remaining in the pool will be higher for the pool.

So, the pool is gaining the “rounding down” error amount with outAmount,
[Code](https://github.com/duality-labs/duality/blob/377592adb4ac0ef445c5e2bc2a73e2635189ccfc/x/dex/keeper/core.go#L299)
Which is eventually, after iterating through liquidity, adding to the totalOut amount.
This amount is sent from the pool to the receiver, upon swapping.

But, Since TruncateInt() is rounding down the calculated amount, it should not be used when calculating the amounts the user needs to deposit or swap into the pool.

So, to sum up:  
if calculating for amountIn, we should round up
if calculating for amountOut, we don't round at all or we round down
    	This is to prevent removing more from the pool than expected due to rounding errors.
